src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/properties/ArmorProperty.java

```diff
@@ -103,26 +114,45 @@ private Builder(int durabilityMultiplier, int[] protectionValues) {
             armorProperty = new ArmorProperty(durabilityMultiplier, protectionValues);
         }
 
+        /**
+         * Set armors made from this Material as unbreakable, bypassing all durability.
+         */
         public ArmorProperty.Builder unbreakable() {
             armorProperty.durabilityMultiplier = 0;
             return this;
         }
 
+        /**
+         * Set the base enchantability of a tool made from this Material. Iron is 14, Diamond is 10, Stone is 5.
+         */
         public ArmorProperty.Builder enchantability(int enchantability) {
             armorProperty.enchantability = enchantability;
             return this;
         }
 
+        /**
+         * Set the protection value for a specific piece of armor made from this Material.
+         */
         public ArmorProperty.Builder protectionValue(ArmorItem.Type type, int value) {
             armorProperty.protectionValues.put(type, value);
             return this;
         }
 
+        /**
+         * Set the protection values for all pieces of armor made from this Material.
+         *
+         * @throws IllegalArgumentException If the provided map does not have a value for all 4 armor pieces.
+         */
         public ArmorProperty.Builder protectionValues(Map<ArmorItem.Type, Integer> protectionValues) {
+            Preconditions.checkArgument(protectionValues != null && protectionValues.size() == 4,
+                    "protectionValues must have 4 entries!");
```

Comment:
This was never validated, which differed from the other calls setting this value